haydenseay@Haydens-MacBook-Pro 1600 % diff -urN argon2/Makefile.old argon2/Makefile
--- argon2/Makefile.old	2019-05-20 05:18:00.000000000 -0400
+++ argon2/Makefile	2020-10-02 09:56:18.000000000 -0400
@@ -19,6 +19,7 @@
 BENCH = bench
 GENKAT = genkat
 ARGON2_VERSION ?= ZERO
+AR ?= ar
 
 # installation parameters for staging area and final installation path
 # Note; if Linux and not Debian/Ubuntu version also add lib override to make command-line
@@ -49,7 +50,7 @@
 				-Wextra -Wno-type-limits -Werror -coverage -DTEST_LARGE_RAM
 
 OPTTARGET ?= native
-OPTTEST := $(shell $(CC) -Iinclude -Isrc -march=$(OPTTARGET) src/opt.c -c \
+OPTTEST := $(shell $(CC) $(CFLAGS) -Iinclude -Isrc -march=$(OPTTARGET) src/opt.c -c \
 			-o /dev/null 2>/dev/null; echo $$?)
 # Detect compatible platform
 ifneq ($(OPTTEST), 0)
@@ -61,9 +62,9 @@
 	SRC += src/opt.c
 endif
 
-BUILD_PATH := $(shell pwd)
-KERNEL_NAME := $(shell uname -s)
-MACHINE_NAME := $(shell uname -m)
+BUILD_PATH ?= $(shell pwd)
+KERNEL_NAME ?= $(shell uname -s)
+MACHINE_NAME ?= $(shell uname -m)
 
 LIB_NAME = argon2
 PC_NAME = lib$(LIB_NAME).pc
@@ -182,7 +183,7 @@
 		$(CC) $(CFLAGS) $(LIB_CFLAGS) $(LDFLAGS) $(SO_LDFLAGS) $^ -o $@
 
 $(LIB_ST): 	$(OBJ)
-		ar rcs $@ $^
+		$(AR) rcs $@ $^
 
 .PHONY: clean
 clean:
